### PR TITLE
Fluffy: Create separate portal rpc handlers for each sub-network

### PR DESCRIPTION
**Pull request by bhartnett**
This PR creates a separate rpc handler for each portal sub-network. 

I've removed the 'network' string parameter and basically copied the code into separate files and renamed a few files and procs. The validation that was added to state store no longer requires the `network == 'state'` check and that code is removed from history and beacon handlers. 

There now is a bit of code duplication which is to be expected and I'm thinking that we can refactor this out in future PRs when we add the network specific validation. Perhaps the duplicated code can go into functions in another file that is shared where it makes sense.

### CODE REVIEW

**Review by kdeme — COMMENTED** [review]
> There now is a bit of code duplication which is to be expected and I'm thinking that we can refactor this out in future PRs when we add the network specific validation. Perhaps the duplicated code can go into functions in another file that is shared where it makes sense.

Yeah, some code could go in functions in a rpc_portal_common or something like that.

However, as I look at this PR now, there are still quite a lot of calls that will never be different between the different subprotocols. E.g. all the ENR related calls. Perhaps we should still keep a generic version around for a part of the calls?

**Comment by bhartnett** [comment]
> > There now is a bit of code duplication which is to be expected and I'm thinking that we can refactor this out in future PRs when we add the network specific validation. Perhaps the duplicated code can go into functions in another file that is shared where it makes sense.
> 
> Yeah, some code could go in functions in a rpc_portal_common or something like that.
> 
> However, as I look at this PR now, there are still quite a lot of calls that will never be different between the different subprotocols. E.g. all the ENR related calls. Perhaps we should still keep a generic version around for a part of the calls?

Yes good point and I guess when adding a new sub-protocol adding the enr related calls would be easier than having to remember to define all the endpoints each time. Okay, I'll make a portal common rpc handler.

File: fluffy/rpc/rpc_portal_beacon_api.nim [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    let
+      key = hexToSeqByte(contentKey)
+      content = hexToSeqByte(contentValue)
+      contentKeys = ContentKeysList(@[ContentKeyByteList.init(key)])
+      numberOfPeers =
+        await p.neighborhoodGossip(Opt.none(NodeId), contentKeys, @[content])
+
+    return numberOfPeers
+
+  rpcServer.rpc("portal_beaconRandomGossip") do(
```
Comment (bhartnett) [inline fluffy/rpc/rpc_portal_beacon_api.nim:261]: @kdeme This endpoint doesn't appear to be in the spec. Should it be moved to a portal_debug namespace or even removed?

File: fluffy/rpc/rpc_portal_debug_history_api.nim [hunk 2]
```diff
@@ -9,20 +9,19 @@
...
 import
   json_rpc/rpcserver,
-  stew/byteutils,
   ../network/wire/portal_protocol,
   ../eth_data/history_data_seeding,
   ../database/content_db
 
 export rpcserver
 
+# TODO: perhaps these endpoints should be named differently staring with "portal_debug_"?
```
Comment (bhartnett) [inline fluffy/rpc/rpc_portal_debug_history_api.nim:18]: Any thoughts on this?

**Comment by bhartnett** [comment]
This PR conflicts with https://github.com/status-im/nimbus-eth1/pull/2689 so I'll wait for that to be merged in before pulling in that code to resolve the conflicts.

File: fluffy/rpc/rpc_portal_beacon_api.nim (hunk 1 above)
Comment (bhartnett) [reply fluffy/rpc/rpc_portal_beacon_api.nim:261]: I see, in that case I'll leave the name and just remove it from the other handlers. Probably we can rename to `portal_beaconGossip` after the spec is updated.

**Review by kdeme — APPROVED** [review]

**Pull request merged**